src/sentry/api/endpoints/internal_register_trusted_relay.py

```diff
@@ -0,0 +1,98 @@
+from datetime import datetime, timezone
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.authentication import UserAuthTokenAuthentication
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.serializers.models.organization import TrustedRelaySerializer
+from sentry.models.options.organization_option import OrganizationOption
+from sentry.utils.audit import create_audit_entry
+
+
+@region_silo_endpoint
+class InternalRegisterTrustedRelayEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.OWNERS_INGEST
+    authentication_classes = (UserAuthTokenAuthentication,)
+
+    def post(self, request: Request, organization) -> Response:
+        """
+        Register a new trusted relay for an organization.
+        If a relay with the given public key already exists, update it.
+        """
+        if not features.has("organizations:relay", organization, actor=request.user):
+            return Response(
+                {"detail": "The organization is not enabled to use an external Relay."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        serializer = TrustedRelaySerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response({"detail": "Invalid request body"}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Get existing trusted relays
+        option_key = "sentry:trusted-relays"
+        try:
+            existing_option = OrganizationOption.objects.get(
+                organization=organization, key=option_key
+            )
+            existing_relays = existing_option.value
+        except OrganizationOption.DoesNotExist:
+            existing_option = None
+            existing_relays = []
```

Comment:
You could also use `OrganizationOption.objects.get_value(organization=organization, key=option_key, default=[])` to read the value without needing a try/except. You wouldn't be able to know if the option was new/existing though.